doc/source/conf.py

```diff
@@ -93,6 +100,80 @@
 
 pio.renderers.default = "sphinx_gallery_png"
 
+
+# add a scikit-image installation step when running in JupyterLite
+
+
+def notebook_modification_function(notebook_content, notebook_filename):
+    notebook_content_str = str(notebook_content)
+    warning_template = "\n".join(
+        [
+            "<div class='alert alert-{message_class}'>",
+            "",
+            "# JupyterLite warning",
+            "",
+            "{message}",
+            "</div>",
+        ]
+    )
+
+    message_class = "warning"
+    message = (
+        "Running the scikit-image examples in JupyterLite is experimental and you may"
+        " encounter some unexpected behaviour.\n\nThe main difference is that imports"
+        " can take a lot longer than usual, for example the first `import skimage`"
+        " statement can take roughly 10-20s.\n\nIf you notice problems, feel free to"
+        " open an [issue](https://github.com/scikit-image/scikit-image/issues/new/choose)."
+    )
+
+    markdown = warning_template.format(message_class=message_class, message=message)
+
+    dummy_notebook_content = {"cells": []}
+    add_markdown_cell(dummy_notebook_content, markdown)
+
+    code_lines = [f"%pip install scikit-image=={version}\n"]
+    code_lines.insert(0, "# JupyterLite-specific code")
+
+    # Extra code lines, dynamically added based on the notebooks' contents.
+    extra_code_lines = []
+    if "plotly" in notebook_content_str:
+        extra_code_lines.append("%pip install plotly")
+    if "matplotlib" in notebook_content_str:
+        extra_code_lines.append("%pip install matplotlib")
+    if "pandas" in notebook_content_str:
+        extra_code_lines.append("%pip install pandas")
+    if "sklearn" in notebook_content_str:
+        extra_code_lines.append("%pip install scikit-learn")
+    if "networkx" in notebook_content_str:
+        extra_code_lines.append("%pip install networkx")
+    if "pywt" in notebook_content_str:
+        extra_code_lines.append("%pip install PyWavelets")
+    if (
+        "from skimage import data" in notebook_content_str
+        or "skimage.data" in notebook_content_str
+        or "ski.data" in notebook_content_str
+        or re.search(
+            r'from\s+skimage\s+import\s+(?:[^,\n]+,\s*)*data(?:\s*,|$)',
+            notebook_content_str,
+        )
```

Comment:
The regex here matches cases where one can write `from skimage import <...some other function>, data`, or arbitrary whitespace, such that cases like `from  skimage        import    data` can also be detected.